infra/gravity/provision.go

```diff
@@ -155,46 +154,47 @@ func (c *TestContext) provisionOps(cfg ProvisionerConfig) ([]Gravity, DestroyFn,
 	destroyFn := cfg.DestroyOpsFn(c, clusterName)
 
 	// monitor the cluster until it's created or times out
-	timeout := time.After(cloudInitTimeout)
-	tick := time.Tick(15 * time.Second)
+	ctx, cancel := context.WithTimeout(c.Context(), cloudInitTimeout)
+	defer cancel()
 
 	c.Logger().Debug("waiting for ops center provisioning to complete")
-Loop:
-	for {
-		select {
-		case <-timeout:
-			return nil, destroyFn, errors.New("clusterInitTimeout exceeded")
-		case <-tick:
-			// check provisioning status
-			status, err := getTeleClusterStatus(clusterName)
-			c.Logger().WithField("status", status).Debug("provisioning status")
-			if err != nil {
-				return nil, destroyFn, trace.Wrap(err)
-			}
+	retryer := wait.Retryer{
+		Delay:    15 * time.Second,
+		Attempts: 1000,
+	}
 
-			switch status {
-			case "installing":
-				// we're still installing, just continue the loop
-			case "active":
-				// the cluster install completed, we can continue the install process
-				break Loop
-			default:
-				return nil, destroyFn, trace.BadParameter("unexpected cluster status: %v", status)
-			}
+	err = retryer.Do(ctx, func() (err error) {
+		// check provisioning status
+		status, err := getTeleClusterStatus(ctx, clusterName)
+		if err != nil {
+			return trace.Wrap(err)
 		}
+		c.Logger().WithField("status", status).Debug("provisioning status")
+
+		switch status {
+		case "installing":
+			// we're still installing, just continue the loop
+			return trace.BadParameter("installation not complete")
```

Comment:
`trace.Retry`?
Most of the time comments are not necessary if other things agree - e.g. logging/error messages giving the same hints. Comments should add value where code is otherwise hard to understand. In other cases they're a distraction.